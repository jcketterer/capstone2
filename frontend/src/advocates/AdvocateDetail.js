import React, { useEffect, useState, useContext, useCallback } from 'react';
import { Redirect, useParams } from 'react-router-dom';
import UserContext from '../UserContext';
import AdvocateAPI from '../api';
import { Link } from 'react-router-dom';
import SkillCard from '../skills/SkillsCard';

const AdvocateDetail = () => {
  const { id } = useParams();
  const user = useContext(UserContext);
  const [advocate, setAdvocate] = useState(true);

  useEffect(
    function getAdvoInfo() {
      async function getAdvocate() {
        setAdvocate(await AdvocateAPI.getAdvocate(id));
      }
      getAdvocate();
    },
    [id]
  );

  let hireDate = advocate.hireDate;
  let hDate = new Date(hireDate);
  let hireConvert = hDate.toDateString();

  let milestoneDate = advocate.current_milestone_start_date;
  let mDate = new Date(milestoneDate);
  let mileConvert = mDate.toDateString();

  return (
    <div className="AdvocateDetail col-md-8 offset-md-2">
      <div className="card">
        <div className="card-header">
          <h2 className="display-4">
            {advocate.firstName} {advocate.lastName}
          </h2>
        </div>
        <div className="card-body" key={advocate.advcoateId}>
          <h3>Email: {advocate.email}</h3>
          <h3>Hire Date: {hireConvert}</h3>
          <h3>Milestone: {advocate.milestone}</h3>
          <h3>Current Milestone Start Date: {mileConvert}</h3>
          <h3>Team Lead: {advocate.teamLead}</h3>
          <h3>Manager: {advocate.manager}</h3>
          <div className="blockquote">
            Skills:
            <ul style={{ listStyleType: 'none' }}>
              {advocate.skills?.map(skill => (
                <li className="lead">{skill.skillName}</li>
              ))}
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AdvocateDetail;
