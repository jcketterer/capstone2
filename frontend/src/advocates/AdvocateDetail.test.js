import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import AdvocateDetail from './AdvocateDetail';
import UserContext from '../UserContext';

it('should render without breaking', () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: 'test' }}>
        <AdvocateDetail />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it('should render and match snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: 'test' }}>
        <AdvocateDetail />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
