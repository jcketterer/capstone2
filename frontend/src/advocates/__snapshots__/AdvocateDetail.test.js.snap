// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="AdvocateDetail col-md-8 offset-md-2"
  >
    <div
      class="card"
    >
      <div
        class="card-header"
      >
        <h2
          class="display-4"
        >
           
        </h2>
      </div>
      <div
        class="card-body"
      >
        <h3>
          Email: 
        </h3>
        <h3>
          Hire Date: Invalid Date
        </h3>
        <h3>
          Milestone: 
        </h3>
        <h3>
          Current Milestone Start Date: Invalid Date
        </h3>
        <h3>
          Team Lead: 
        </h3>
        <h3>
          Manager: 
        </h3>
        <div
          class="blockquote"
        >
          Skills:
          <ul
            style="list-style-type: none;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
